p2p/src/lib.rs

```diff
@@ -203,7 +214,36 @@ pub async fn make_p2p<S: PeerDbStorage + 'static>(
         p2p_config.socks5_proxy.is_some(),
     )?;
 
-    if let Some(socks5_proxy) = &p2p_config.socks5_proxy {
+    if let Some(true) = p2p_config.disable_noise {
+        ensure!(
+            *chain_config.chain_type() == ChainType::Regtest,
+            P2pError::InvalidConfigurationValue(
+                "P2P encryption can only be disabled on the regtest network".to_owned()
+            )
+        );
+        ensure!(
+            p2p_config.socks5_proxy.is_none(),
+            P2pError::InvalidConfigurationValue(
+                "SOCKS5 proxy support is not implemented for plaintext".to_owned()
```

Comment:
plaintext? I'm not sure I understand what this means.

Reply:
Oh... you mean unencrypted.

Please call it unencrypted.

Reply:
Sure, updated.